### PR TITLE
XGBoost read data from Maxcompute Using PAI-TF API

### PR DESCRIPTION
**Pull request by Yancey0623**
related issue: #1819 
This PR makes XGBoost read data from Maxcmpute Using PAI-TF API, so that we can run XGBoost via AlisaSubmitter on Dataworks.

### CODE REVIEW

File: pkg/sql/codegen/xgboost/template_pred.go [hunk 1]
```diff
@@ -44,7 +46,10 @@ pred(datasource='''{{.DataSource}}''',
...
      hive_location='''{{.HiveLocation}}''',
      hdfs_user='''{{.HDFSUser}}''',
-     hdfs_pass='''{{.HDFSPass}}''')
+		 hdfs_pass='''{{.HDFSPass}}''',
+		 is_pai="{{.IsPAI}}" == "true",
+		 pai_table="{{.PAITable}}"
+		 )
 `
 
 var predTemplate = template.Must(template.New("Pred").Parse(predTemplateText))
```
Comment (weiguoz) [inline pkg/sql/codegen/xgboost/template_pred.go:61]: indent?

File: python/sqlflow_submitter/xgboost/train.py [hunk 2]
```diff
@@ -13,14 +13,30 @@
...
+                pai_table=""):
+
     if label_spec:
         label_spec["feature_name"] = label_spec["name"]
-    gen = db_generator(conn.driver, conn, dataset_sql, feature_column_name,
-                       label_spec, feature_spec)
+    feature_column_name = [k["name"] for k in feature_field_meta]
+    feature_spec = {k['name']: k for k in feature_field_meta}
+
+    if is_pai:
```
Comment (typhoonzero) [inline python/sqlflow_submitter/xgboost/train.py:29]:
```
if is_pai: 
    gen = pai_maxcompute_input_fn(...
```

**Comment by typhoonzero** [comment]
By the way we can add alisa test cases in the internal CI system.

**Comment by typhoonzero** [comment]
The title of this PR should be xgboost can read from MaxCompute using PAI TF APIs.

**Comment by Yancey0623** [comment]
> By the way we can add Alisa test cases in the internal CI system.

@typhoonzero Good idea, I added some Alisa e2e test cases and would keep improving them.

File: pkg/sql/codegen/pai/template_xgboost.go [hunk 3]
```diff
@@ -27,8 +27,9 @@ model.save_metas("{{.OSSModelDir}}",
 		   "", # estimator = ""
            model_params,
            train_params,
-           feature_field_meta,
-           label_field_meta)
+					 feature_metas,
```
Comment (typhoonzero) [inline pkg/sql/codegen/pai/template_xgboost.go:30]: Indentation problem?

File: python/sqlflow_submitter/xgboost/predict.py [hunk 4]
```diff
@@ -20,24 +20,23 @@
...
          is_pai=False,
          hdfs_namenode_addr="",
          hive_location="",
          hdfs_user="",
-         hdfs_pass=""):
+         hdfs_pass="",
+         pai_table=""):
     # TODO(typhoonzero): support running on PAI without MaxCompute AK/SK connection.
-    conn = connect_with_data_source(datasource)
+    if not is_pai:
```
Comment (typhoonzero) [inline python/sqlflow_submitter/xgboost/predict.py:34]: Can remove above TODO comment now.

**Review by typhoonzero — APPROVED** [review]
LGTM

File: python/sqlflow_submitter/xgboost/predict.py (hunk 4 above)
Comment (Yancey0623) [reply python/sqlflow_submitter/xgboost/predict.py:34]: Will do that in the next PR.

**Pull request merged**